From a2bf08b5e563e85ff4cf4862d08148d6c7584979 Mon Sep 17 00:00:00 2001
From: Gustav Behm <me@rootmos.io>
Date: Sat, 30 Nov 2019 14:13:03 +0100
Subject: [PATCH] Handle non-existent relative directory data/genesis_txs

---
 src/ar_weave.erl | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/src/ar_weave.erl b/src/ar_weave.erl
index 88863d1..1152ac3 100644
--- a/src/ar_weave.erl
+++ b/src/ar_weave.erl
@@ -378,15 +378,17 @@ is_tx_on_block_list([#block { txs = TXs }|Bs], TXID) ->
 is_data_on_block_list(_, _) -> false.
 
 read_genesis_txs() ->
-	{ok, Files} = file:list_dir("data/genesis_txs"),
-	lists:foldl(
-		fun(F, Acc) ->
-			file:copy("data/genesis_txs/" ++ F, ar_meta_db:get(data_dir) ++ "/" ++ ?TX_DIR ++ "/" ++ F),
-			[ar_util:decode(hd(string:split(F, ".")))|Acc]
-		end,
-		[],
-		Files
-	).
+    case file:list_dir("data/genesis_txs") of
+	    {ok, Files} -> lists:foldl(
+                         fun(F, Acc) ->
+                                 file:copy("data/genesis_txs/" ++ F, ar_meta_db:get(data_dir) ++ "/" ++ ?TX_DIR ++ "/" ++ F),
+                                 [ar_util:decode(hd(string:split(F, ".")))|Acc]
+                         end,
+                         [],
+                         Files
+                        );
+        {error, _} -> []
+    end.
 
 create_genesis_txs() ->
 	TXs = lists:map(
-- 
2.24.0

